core/src/main/scala/org/apache/spark/scheduler/SchedulerBackend.scala

```diff
@@ -92,4 +93,16 @@ private[spark] trait SchedulerBackend {
    */
   def maxNumConcurrentTasks(rp: ResourceProfile): Int
 
+  /**
+   * Get the list of both active and dead executors host locations for push based shuffle
```

Comment:
How about
```suggestion
   * Get the list of host locations for push based shuffle
```
?

We actually can not know the host of a dead executor in Spark since we'd always remove the related info of that dead executor.